### ISSUE TITLE
GUI: Creating Proper Tests for Inventory Structs

### ISSUE THREAD

**Issue by ukanyanwu** [uchicago-cs/chiventure#1459]
Many of the modules within graphics have their own structs that are used to make up different modules ,such as inventory, npcs, or scenes, that become heap allocated and initialized. Within inventory_graphics, this includes the player_inventory struct. This struct's new and free function need to be properly tested to make sure that they allocate and initialize memory correctly, and said memory is freed correctly, as well. These tests will then allow us to make any changes to these functions as needed.

**Comment by ukanyanwu** [comment]
Currently, there are tests written in a main .c file for the player_inventory struct. Unfortunately, player_inventory relies on using item.h within game-state for one of its parameters, and there are issues with it not being recognized and therefore unable to compile. @dhirpalshah is also looking into this issue in order to fully verify that the new and free functions for player_inventory actually print the proper values.

**Comment by ukanyanwu** [comment]
Tests are currently being created for the Inventory Module, however we've discovered that some of our functions are unable to run because they rely on indexing the 2D array that represents the inventory with the x and y positions of an item. However, under item.h, the x and y positions for items are pointers, meaning that they cannot be used to index the array the way we wish to in important functions for the inventory. @dhirpalshah is working on how to get these fields changed to regular unsigned ints rather than pointer so that the functions can compile and allow us to properly test.

**Comment by dhirpalshah** [comment]
I have contacted @KendrickXie from team action management to changed these fields, so the pull request for their part should be in ASAP.

**Comment by ukanyanwu** [comment]
With the changes to item.h made, inventory_graphics can finally be ready for tests. However, at a meeting with @csraffaga 
 today, creating tests has revealed that   new/init/free functions within the inventory module need to be created, along with the additional functions specified in #1618 . Once these functions are remodeled, they will be able to make a pass of tests. The tests will be able to point out further issues with these functions, but it's important to make the functions usable when that time comes.

**Comment by ukanyanwu** [comment]
We currently have tests for inventory slot_t structs. Unfortunately there are issues with building the files that contain these tests. Despite linking to the right directories and files as far as we know, the tests are not recognizing functions unique to the items module such as item_new and get_all_items_in_inventory, this is a problem as some of our tests/functions rely on these functions in order to work properly. Recently, @imadejski has been receiving help from @philipchenaux in order to solve this issue. Apparently this may not be a universal problem across machines, but it's still important to minimize where this problem appears so that developments can be made on the inventory during this sprint and future iterations of this class.

**Comment by ukanyanwu** [comment]
We've resolved the issues we faced for inventory_graphics tests, and inventory_graphics are now merged with the dev branch. Tests for the structs within inventory_graphics are written properly and pass, giving us confidence that our new/init/free functions that were written for inventory structs were written properly. The written tests allow further information on a deeper end of how these functions work, allowing an added level of information to the inventory module when approached again in the future.
